cli/file_fetcher.rs

```diff
@@ -2,677 +2,499 @@
-    let file_fetcher = Self {
-      http_cache,
-      source_file_cache: SourceFileCache::default(),
-      cache_blocklist,
-      use_disk_cache,
-      no_remote,
-      cached_only,
-      http_client: create_http_client(ca_file)?,
-    };
-
-    Ok(file_fetcher)
-  }
-
-  pub fn check_if_supported_scheme(url: &Url) -> Result<(), AnyError> {
-    if !SUPPORTED_URL_SCHEMES.contains(&url.scheme()) {
-      return Err(generic_error(format!(
-        "Unsupported scheme \"{}\" for module \"{}\". Supported schemes: {:#?}",
-        url.scheme(),
-        url,
-        SUPPORTED_URL_SCHEMES
-      )));
+impl Cache {
+  /// Returns if the cache should be used for a given specifier.
+  pub fn should_use(&self, specifier: &ModuleSpecifier) -> bool {
+    match self {
+      Cache::ReloadAll => false,
+      Cache::Use | Cache::Only => true,
+      Cache::ReloadSome(list) => {
+        let mut url = specifier.as_url().clone();
+        url.set_fragment(None);
+        if list.contains(&url.as_str().to_string()) {
+          return false;
+        }
+        url.set_query(None);
+        let mut path = PathBuf::from(url.as_str());
+        loop {
+          if list.contains(&path.to_str().unwrap().to_string()) {
+            return false;
+          }
+          if !path.pop() {
+            break;
+          }
+        }
+        true
+      }
     }
-
-    Ok(())
   }
+}
 
-  /// Required for TS compiler and source maps.
-  pub fn fetch_cached_source_file(
-    &self,
-    specifier: &ModuleSpecifier,
-    permissions: Permissions,
-  ) -> Option<SourceFile> {
-    let maybe_source_file = self.source_file_cache.get(specifier.to_string());
-
-    if maybe_source_file.is_some() {
-      return maybe_source_file;
-    }
-
-    // If file is not in memory cache check if it can be found
-    // in local cache - which effectively means trying to fetch
-    // using "--cached-only" flag.
-    // It should be safe to for caller block on this
-    // future, because it doesn't actually do any asynchronous
-    // action in that path.
-    if let Ok(maybe_source_file) =
-      self.get_source_file_from_local_cache(specifier.as_url(), &permissions)
-    {
-      return maybe_source_file;
-    }
+/// Fetch a source file from the local file system.
+fn fetch_local(specifier: &ModuleSpecifier) -> Result<File, AnyError> {
+  let local = specifier.as_url().to_file_path().map_err(|_| {
+    uri_error(format!("Invalid file path.\n  Specifier: {}", specifier))
+  })?;
+  let bytes = fs::read(local.clone())?;
+  let charset = text_encoding::detect_charset(&bytes).to_string();
+  let source = get_source_from_bytes(bytes, Some(charset))?;
+  let media_type = MediaType::from(specifier);
+
+  Ok(File {
+    local,
+    maybe_types: None,
+    media_type,
+    source,
+    specifier: specifier.clone(),
+  })
+}
 
-    None
+/// Return a validated scheme for a given module specifier.
+fn get_scheme(specifier: &ModuleSpecifier) -> Result<String, AnyError> {
```

Comment:
`validate_scheme`?

Reply:
that would imply that it is returning a boolean or something... `get_validated_scheme` I guess...